jaeger-core/src/main/java/com/uber/jaeger/Configuration.java

```diff
@@ -263,6 +264,354 @@ public void setStatsFactory(StatsFactory statsFactory) {
+     * @return the same builder's instance
+     */
+    public Builder withCodecConfiguration(CodecConfiguration codecConfiguration) {
+      this.codecConfiguration = codecConfiguration;
+      return this;
+    }
+
+    /**
+     * Sets the {@link StatsFactory} to use. If {@link #withStatsReporter(StatsReporter)} has been called already,
+     * its outcome is overridden by the parameter provided to this method.
+     * @see StatsFactory
+     * @see #withStatsReporter(StatsReporter)
+     * @param statsFactory the StatsFactory to use
+     * @return the same builder's instance
+     */
+    public Builder withStatsFactory(StatsFactory statsFactory) {
+      this.statsFactory = statsFactory;
+      return this;
+    }
+
+    /**
+     * Creates a new {@link StatsFactory} based on the given {@link StatsReporter}. If
+     * {@link #withStatsFactory(StatsFactory)} has been called already, it gets overridden by the {@link StatsFactory}
+     * created by this method.
+     * @see StatsReporter
+     * @see #withStatsFactory(StatsFactory)
+     * @param statsReporter the StatsReporter to use
+     * @return the same builder's instance
+     */
+    public Builder withStatsReporter(StatsReporter statsReporter) {
+      this.statsFactory = new StatsFactoryImpl(statsReporter);
+      return this;
+    }
+
+    /**
+     * Sets the sampler type and parameter to use when constructing a new {@link SamplerConfiguration}. If a
+     * {@link SamplerConfiguration} has been set already, it will be overridden.
+     *
+     * @see SamplerConfiguration#param
+     * @see SamplerConfiguration#type
+     * @param type  the sampler type
+     * @param param the sampler parameter
+     * @return the same builder's instance
+     */
+    public Builder withSamplerTypeAndParam(String type, Number param) {
+      this.samplerConfiguration = null;
+      this.samplerType = type;
+      this.samplerParameter = param;
+      return this;
+    }
+
+    /**
+     * Sets the sampler manager host and port to use when constructing a new {@link SamplerConfiguration}. If a
+     * {@link SamplerConfiguration} has been set already, it will be overridden. This is only effective if a sampler
+     * type and parameter have been specified as well.
+     *
+     * @param samplerManagerHostPort the manager host:port, like "localhost:5778"
+     * @return the same builder's instance
+     */
+    public Builder withSamplerManagerHostPort(String samplerManagerHostPort) {
+      this.samplerManagerHostPort = samplerManagerHostPort;
+      return this;
+    }
+
+    /**
+     * Sets the codecs based on the given propagation format. If a {@link CodecConfiguration} has been set already, it
+     * will be overridden by this.
+     *
+     * @see CodecConfiguration#fromPropagationString(String)
+     * @param propagationFormat the propagation format
+     * @return the same builder's instance
+     */
+    public Builder withPropagationFormat(String propagationFormat) {
```

Comment:
Just looked into adding this, but the code actually accepts *more than one* format. It can be specified as: "jaeger,b3", and it will add appropriate codecs for Jaeger and B3.

The formats are documented under usual place for the doc on env vars (https://github.com/jaegertracing/jaeger-client-java/tree/master/jaeger-core), but I'll add a javadoc to `CodecConfiguration#fromPropagationString` as well.

Reply:
>  It can be specified as: "jaeger,b3", and it will add appropriate codecs for Jaeger and B3.

I think this is mostly the case when the configuration is created from env vars. here it could accept a list or injectors/extractors

Reply:
I would rather keep it close to the underlying configuration. Also, it might be easier for the simple case to just accept a list of `Propagation`. Anything more complex can be done by using the `CodecConfiguration` + `withCodecConfiguration`.